watchdog/src/http.rs

```diff
@@ -0,0 +1,67 @@
+use ic_cdk::api::management_canister::http_request::{
+    CanisterHttpRequestArgument, HttpHeader, HttpResponse, TransformArgs,
+};
+
+pub type TransformFn = fn(TransformArgs) -> HttpResponse;
+
+/// Stores the configuration of the HTTP request, which includes:
+/// - URL
+/// - HTTP request
+/// - Transform function inner implementation to be called inside the canister's endpoint
+pub struct HttpRequestConfig {
+    // URL is added only for tests.
+    #[allow(dead_code)]
+    url: String,
+
+    request: CanisterHttpRequestArgument,
+    transform_implementation: TransformFn,
+}
+
+impl HttpRequestConfig {
+    pub fn new<T>(
+        url: &str,
+        transform_endpoint: Option<T>,
+        transform_implementation: TransformFn,
+    ) -> Self
+    where
+        T: Fn(TransformArgs) -> HttpResponse + 'static,
+    {
+        Self {
+            url: String::from(url),
+            request: create_request(url, transform_endpoint),
+            transform_implementation,
+        }
+    }
+
+    // URL is added only for tests.
+    #[allow(dead_code)]
+    pub fn url(&self) -> &str {
+        &self.url
+    }
```

Comment:
nits:
1. If something is needed for tests, it's clearer to use the `cfg(test)` directive as opposed to marking as dead code.
2. Do you actually need to store the url? Does this suffice?

```
    #[cfg(test)]
    pub fn url(&self) -> &str {
        &self.request.url
    }
```

Reply:
Great idea, thank you! Done.